server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -288,15 +295,36 @@ public String execute(ReplicationCoordinator.Client client) throws Exception {
             }
           } else {
             span = Trace.start("WAL replication");
+
+            ExecutorService executor = Executors.newFixedThreadPool(1);
```

Comment:
Any reason that we need to start an extra thread here?

Guava has a `MoreExecutors` class with a special Executor that uses the same thread. Would that work here?